migration_steps/validation/validate_db/app/app.py

```diff
@@ -96,11 +69,178 @@ def main(verbose):
+        sirius_table_name = get_sirius_table(mapping)
+        map_dict = get_map_dict(mapping)
+
+        casrec_cols = []
+        casrec_tables = []
+        for k, v in map_dict.items():
+            if v["transform_casrec"]["casrec_table"]:
+                casrec_col_table = v["transform_casrec"]["casrec_table"].lower()
+                casrec_col_name = v["transform_casrec"]["casrec_column_name"]
+                casrec_tables.append(config.schemas['casrec_csv'] + '.' + casrec_col_table)
+                col = f'{casrec_col_table}."{casrec_col_name}"'
+            elif '' != v["transform_casrec"]["default_value"]:
+                col = format_default_value(v)
+            elif '' != v["transform_casrec"]["calculated"]:
+                col = format_calculated_value(v)
+            casrec_cols.append(
+                f"{indent}{indent}{indent}{casrec_col_sql(v, col)} AS {k}"
+            )
+
+        separator = ',\n'
+        casrec_cols = separator.join(casrec_cols)
+        sirius_cols = separator.join([f"{indent}{indent}{indent}{s} AS {s}" for s in map_dict.keys()])
+
+        separator = ','
+        casrec_tables = separator.join(set(casrec_tables))
+
+        sql_lines.append(f"INSERT INTO {exception_table_name}(\n")
+        # casrec half
+        sql_lines.append(f"{indent}SELECT * FROM(\n")
+        sql_lines.append(f"{indent}{indent}SELECT DISTINCT\n")
+        sql_lines.append(f"{casrec_cols}\n")
+        sql_lines.append(f"{indent}{indent}FROM {casrec_tables}\n")
+        sql_lines.append(f"{indent}{indent}ORDER BY caserecnumber ASC\n")
+        # sql_lines.append(f"{indent}{indent}LIMIT 10\n")
+        sql_lines.append(f"{indent}) as csv_data\n")
+        sql_lines.append(f"{indent}EXCEPT\n")
+        # sirius half
+        sql_lines.append(f"{indent}SELECT * FROM(\n")
+        sql_lines.append(f"{indent}{indent}SELECT DISTINCT\n")
+        sql_lines.append(sirius_cols + "\n")
+        sql_lines.append(f"{indent}{indent}FROM {sirius_table_name}\n")
+        sql_lines.append(f"{indent}{indent}WHERE clientsource = 'CASRECMIGRATION'\n")
+        sql_lines.append(f"{indent}{indent}ORDER BY caserecnumber ASC\n")
+        # sql_lines.append(f"{indent}{indent}LIMIT 10\n")
+        sql_lines.append(f"{indent}) as sirius_data\n")
+        sql_lines.append(");\n\n")
+
+
+def build_results_sql():
+    sql_file = open(sql_path / "get_validation_results.sql", 'w')
+    reported_mappings = []
+    for mapping in mappings_to_run:
+        exception_table_name = get_exception_table(mapping)
+        reported_mappings.append(f"SELECT '{mapping}', (SELECT count(*) FROM {exception_table_name})\n")
+    separator = 'UNION\n'
+    sql = separator.join(reported_mappings)
+    sql_file.writelines(sql)
+    sql_file.close()
+
+
+def show_results(conn_target):
+    build_results_sql()
+    results_df = df_from_sql_file(
+        sql_path, "get_validation_results.sql", conn_target
+    )
+    print(
```

Comment:
Can you swap `print` statements for logs? Gives us more control over what data appears where.

Reply:
Done